src/NUnitFramework/framework/Attributes/RangeAttribute.cs

```diff
@@ -23,15 +23,28 @@
 
 using System;
 using System.Collections;
+using System.Reflection;
+
+using NUnit.Compatibility;
+using NUnit.Framework.Interfaces;
 
 namespace NUnit.Framework
 {
     /// <summary>
     /// RangeAttribute is used to supply a range of values to an
     /// individual parameter of a parameterized test.
     /// </summary>
-    public class RangeAttribute : ValuesAttribute
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = true)]
+    public class RangeAttribute : DataAttribute, IParameterDataSource
     {
+        /// <summary>
+        /// The collection of data to be returned.
+        /// We use an object[] so that the individual
+        /// elements may have their type changed in GetData
+        /// if necessary
+        /// </summary>
+        // TODO: This causes a lot of boxing so we should eliminate it
+        protected object[] data;
```

Comment:
This can now be private.

Reply:
Alright, cool. I wasn't sure if I could do that or if I needed to leave it protected in case another attribute needed to derive from `RangeAttribute` in the future. 

I went ahead and renamed this to `_data` while I was at it to be more consistent with the coding standards. It's the only private field in this class, so it seemed like a safe time to make that change, but if it's not I can back it out.